057-run-zk-kraft-clusters-parallel.md

```diff
@@ -0,0 +1,65 @@
+# Allow running ZooKeeper and KRaft based clusters in parallel
+
+This proposal is about changing the way the current `UseKRaft` feature gate works.
+It changes the way the Strimzi Cluster Operator handles ZooKeeper and KRaft based clusters when it is enabled.
+
+## Current situation
+
+Currently, when the `UseKRaft` feature gate is enabled (together with the required `KafkaNodePools` one), the operator expects all the already running Apache Kafka clusters being KRaft-based.
+It means that the `Kafka` custom resources are configured together with `KafkaNodePool`(s) with `broker` and `controller` roles.
+There is no way to differentiate between clusters running in ZooKeeper or KRaft mode.
+When a `Kafka` custom resource is configured to actually use ZooKeeper or it's just badly configured but supposed to be KRaft based, the operator detects it as having a missing KRaft controllers configuration and logs the following warning:
+
+```shell
+io.strimzi.operator.common.model.InvalidResourceException: Tha Kafka cluster my-cluster is invalid: [At least one KafkaNodePool with the controller role and at least one replica is required when KRaft mode is enabled]
```

Comment:
```suggestion
io.strimzi.operator.common.model.InvalidResourceException: The Kafka cluster my-cluster is invalid: [At least one KafkaNodePool with the controller role and at least one replica is required when KRaft mode is enabled]
```

Reply:
Maybe you should open a PR to fix it in the code instead? 😮

Reply:
Yeah, I will update here and open the PR on code.

Reply:
PR is here https://github.com/strimzi/strimzi-kafka-operator/pull/9149